junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/discovery/DefaultClassDescriptor.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2015-2020 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * https://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.jupiter.engine.discovery;
+
+import java.lang.annotation.Annotation;
+import java.util.List;
+import java.util.Optional;
+
+import org.junit.jupiter.api.ClassDescriptor;
+import org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor;
+import org.junit.platform.commons.util.AnnotationUtils;
+import org.junit.platform.commons.util.Preconditions;
+import org.junit.platform.commons.util.ToStringBuilder;
+
+/**
+ * Default implementation of {@link ClassDescriptor}, backed by
+ * a {@link ClassBasedTestDescriptor}.
+ *
+ * @since 5.8
+ */
+class DefaultClassDescriptor implements ClassDescriptor {
+
+	private final ClassBasedTestDescriptor testDescriptor;
+
+	DefaultClassDescriptor(ClassBasedTestDescriptor testDescriptor) {
+		this.testDescriptor = testDescriptor;
+	}
+
+	ClassBasedTestDescriptor getTestDescriptor() {
+		return testDescriptor;
+	}
+
+	@Override
+	public final Class<?> getTestClass() {
+		return this.testDescriptor.getTestClass();
+	}
+
+	@Override
+	public final String getDisplayName() {
+		return this.testDescriptor.getDisplayName();
+	}
+
+	@Override
+	public boolean isAnnotated(Class<? extends Annotation> annotationType) {
+		Preconditions.notNull(annotationType, "annotationType must not be null");
+		return AnnotationUtils.isAnnotated(getTestClass(), annotationType);
+	}
+
+	@Override
+	public <A extends Annotation> Optional<A> findAnnotation(Class<A> annotationType) {
+		Preconditions.notNull(annotationType, "annotationType must not be null");
+		return AnnotationUtils.findAnnotation(getTestClass(), annotationType);
+	}
+
+	@Override
+	public <A extends Annotation> List<A> findRepeatableAnnotations(Class<A> annotationType) {
+		Preconditions.notNull(annotationType, "annotationType must not be null");
+		return AnnotationUtils.findRepeatableAnnotations(getTestClass(), annotationType);
+	}
```

Comment:
Let's avoid the duplication with `DefaultMethodDescriptor`, by extracting a base class that works with `AnnotatedElement`.